arbor/util/dylib.cpp

```diff
@@ -0,0 +1,59 @@
+#include <fstream>
+#include <string>
+
+#include <dlfcn.h>
+
+#include <arbor/arbexcept.hpp>
+
+#include "util/dylib.hpp"
+#include "util/strprintf.hpp"
+
+namespace arb {
+namespace util {
+
+struct dl_handle {
```

Comment:
This is redundant now, it was meant as an opaque handle, but if it's not exposed, it can go.